testrepository/commands/load.py

```diff
@@ -126,35 +134,38 @@ def wrap_result(result):
                 yield (case, str(pos))
-        case = testtools.ConcurrentStreamTestSuite(make_tests)
-        # One unmodified copy of the stream to repository storage
-        inserter = repo.get_inserter(partial=self.ui.options.partial)
-        # One copy of the stream to the UI layer after performing global
-        # filters.
-        try:
-            previous_run = repo.get_latest_run()
-        except KeyError:
-            previous_run = None
-        output_result, summary_result = self.ui.make_result(
-            inserter.get_id, testcommand, previous_run=previous_run)
-        result = testtools.CopyStreamResult([inserter, output_result])
-        runner_thread = None
-        result.startTestRun()
-        try:
-            # Convert user input into a stdin event stream
-            interactive_streams = list(self.ui.iter_streams('interactive'))
-            if interactive_streams:
-                case = InputToStreamResult(interactive_streams[0])
-                runner_thread = threading.Thread(
-                    target=case.run, args=(result,))
-                runner_thread.daemon = True
-                runner_thread.start()
-            case.run(result)
-        finally:
-            result.stopTestRun()
-            if interactive_streams and runner_thread:
-                runner_thread.stop = True
-                runner_thread.join(10)
+        # Setup testcommand to query profiles etc.
+        with testcommand:
+            case = testtools.ConcurrentStreamTestSuite(make_tests)
+            # One unmodified copy of the stream to repository storage
+            inserter = repo.get_inserter(
+                partial=self.ui.options.partial, profiles=testcommand.profiles)
+            # One copy of the stream to the UI layer after performing global
+            # filters.
+            try:
+                previous_run = repo.get_latest_run()
+            except KeyError:
+                previous_run = None
+            output_result, summary_result = self.ui.make_result(
+                inserter.get_id, testcommand, previous_run=previous_run)
+            result = testtools.CopyStreamResult([inserter, output_result])
+            runner_thread = None
+            result.startTestRun()
+            try:
+                # Convert user input into a stdin event stream
+                interactive_streams = list(self.ui.iter_streams('interactive'))
+                if interactive_streams:
+                    case = InputToStreamResult(interactive_streams[0])
+                    runner_thread = threading.Thread(
+                        target=case.run, args=(result,))
+                    runner_thread.daemon = True
+                    runner_thread.start()
+                case.run(result)
+            finally:
+                result.stopTestRun()
+                if interactive_streams and runner_thread:
+                    runner_thread.stop = True
+                    runner_thread.join(10)
```

Comment:
I wish Github made it easier to figure out whether this code was changed beyond mere indentation. 

Can I interest you in trying reviewable.io?